### ISSUE TITLE
Door open the wrong direction half the time (Linux 32 bit)

### ISSUE THREAD

**Issue by ETLegacyTeam** [etlegacy/etlegacy#1206]
Author: @harleking (Harlekin)
Date: 2019-02-13
Redmine Issue: 1206, https://dev.etlegacy.com/issues/1206

---

Noticed that doors open to the wrong direction, if i try to open them crouched. eg: Radar the door at the gate opens towards the player if i come from the axis bracks side.

---

Related issue(s): #1095

**Comment by ETLegacyTeam** [comment]
From: @rmarquis (Spyhawk)
Date: 2019-02-14T15:45:52Z

---

Cannot reproduce. Are you sure somebody else (bot?) didn't open the door from the other side?

**Comment by ETLegacyTeam** [comment]
From: @IR4T4 (IR4T4)
Date: 2019-02-14T17:14:33Z

---

Axis team door near big gate on radar seems to have issues. The direction of opening is alternating.

How to reproduce:
Spawn as axis and open the door from gate side only.

**Comment by ETLegacyTeam** [comment]
From: @rmarquis (Spyhawk)
Date: 2019-02-14T18:47:29Z

---

Which "gate" are we talking about? The gate the allies have to dynamite, near the truck that collects the radar parts? I still can't reproduce.

**Comment by ETLegacyTeam** [comment]
From: @harleking (Harlekin)
Date: 2019-02-14T19:11:08Z

---

It is viewangle related. i add a demo to show it.

**Comment by ETLegacyTeam** [comment]
From: @rmarquis (Spyhawk)
Date: 2019-02-14T20:49:42Z

---

It looks like the issue is platform specific. It doesn't happen on windows (localhost) but it looks like it does when qagame is hosted on a Linux machine (?).

**Comment by ETLegacyTeam** [comment]
From: @rmarquis (Spyhawk)
Date: 2019-03-09T17:34:12Z

---

The issue seems to appear on Linux 32 bit only. This is likely related to a math library issue (ie, check MAX_PATCH_PLANE related code).

**Comment by rmarquis** [comment]
In light of #1095 being fixed, I wonder if that issue still occurs. Anyone with a 32bit Linux laying around caring enough to have a look?

**Comment by Aciz** [comment]
Yes, this still occurs on my 32-bit 2.76 server. However, it did not occur when running ETJump, only with Legacy mod.

Edit: should note that the legacy mod version used in testing was 2.76 also

**Comment by donishki** [comment]
I leave this here, so it doesn't get lost in discord:

https://streamable.com/2yn50z
https://streamable.com/u8v40r

The server was running on 32bit Linux. The system specs are the following:

https://pastebin.com/YvYJd8N7

Moving to the 64bit Linux version fixed the problem for me.

**Comment by Aciz** [comment]
Still an issue with 2.78, and as I tested earlier, seems to only apply to Legacy mod specifically, on 32-bit Linux.

**Comment by rmarquis** [comment]
As noted in #1095, this could be the result of some newer compilers options (fast math or something), not necessarily related to any code change directly.

I'd also consider this a non-issue: Linux 32 bit being pushed out of existence, and there are virtually no reason to install a Linux 32 server to run Legacy mod, it's only useful for running old third party mods that shouldn't be affected anyway (btw, we should add something about that in the FAQ).
